### PR TITLE
Change test go dep to go mod

### PR DESCRIPTION
**Pull request by lberk**
As requested in #521 -- making a separate PR for just the go.mod changes

### CODE REVIEW

File: test/go.mod [hunk 1]
```diff
@@ -0,0 +1,100 @@
+module github.com/openshift-knative/serverless-operator/test
+
+go 1.14
+
+require (
```
Comment (markusthoemmes) [inline test/go.mod:5]: Please run `go mod tidy` on this. There's way too much stuff in here.

File: test/go.mod [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	knative.dev/test-infra v0.0.0-20200519161858-554a95a37986
+	sigs.k8s.io/yaml v1.2.0 // indirect
+)
+
+replace (
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20200527184302-a843dc3262a0
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20191125132246-f6563a70e19a
+
+	// Pin to kube 1.16
+	k8s.io/api => k8s.io/api v0.0.0-20190918155943-95b840bb6a1f
```
Comment (markusthoemmes) [inline test/go.mod:76]: Can we use actual tags here, that are readable? Do we need to pin all of those?

Comment (lberk) [reply test/go.mod:76]: I couldn't get it working without the pins (which came from olm itself)

https://github.com/operator-framework/operator-lifecycle-manager/blob/release-4.3/go.mod#L54

Comment (lberk) [reply test/go.mod:76]: keep in mind -- it's not serving that made this a royal pain, it was adding eventing that brought in a dependency that wasn't compatible with go mod

File: test/go.mod (hunk 1 above)
Comment (lberk) [reply test/go.mod:5]: So, just to make sure I understand what you're asking for (and the implications).

You'd like a reduced subset of deps in go.{mod,sum} via running `go mod tidy`

However, that'll remove all the eventing bits from #521, which is fine, I'll update the title of *this* PR accordingly, but it'll make it less of a cherry pick and I'll have to rebase 521.

I understand this makes the individual PRs cleaner, just want to make sure we're on the same page and you'd still like that done here (vs in #521 where I can run the same `go mod tidy`)

either way, let me know what you'd prefer

Comment (markusthoemmes) [reply test/go.mod:5]: Just seen this comment now. Let's invert the order then. Let's merge this one as-is (if CI is happy) and then the tests.

Comment (lberk) [reply test/go.mod:5]: was asked to do this change here -- I'll rebased #521 as needed.

**Comment by lberk** [comment]
@markusthoemmes changes applied ask, please take a look

File: test/go.mod (hunk 1 above)
Comment (lberk) [reply test/go.mod:5]: @markusthoemmes just saw our comments collided -- could we please get this change in first? I've already started the work to rebase on this PR, as you suggested earlier/elsewhere, it's cleaner to make these changes here, then the eventing PR on top

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**